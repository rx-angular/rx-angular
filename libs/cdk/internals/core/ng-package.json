{
  "$schema": "../../../../node_modules/ng-packagr/package.schema.json",
  "lib": {
    "entryFile": "src/index.ts",
    "flatModuleFile": "cdk-internals-core"
  }
}
