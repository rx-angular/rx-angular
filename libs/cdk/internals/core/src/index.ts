export { getZoneUnPatchedApi } from './lib/get-zone-unpatched-api';
export { timeoutSwitchMapWith } from './lib/timeout';
export { accumulateObservables } from './lib/accumulateObservables';
export { ObservableAccumulation, ObservableMap } from './lib/model';
