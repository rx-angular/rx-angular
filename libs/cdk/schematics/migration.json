{
  "$schema": "../../../node_modules/@angular-devkit/schematics/collection-schema.json",
  "schematics": {
    "update-1.0.0-beta.1": {
      "version": "1.0.0-beta.1",
      "description": "Migrate to sub-entry points",
      "factory": "./src/migrations/update-1.0.0-beta.1/index"
    }
  }
}
