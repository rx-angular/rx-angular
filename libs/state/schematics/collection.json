{
  "$schema": "../../../node_modules/@angular-devkit/schematics/collection-schema.json",
  "schematics": {
    "ng-add": {
      "aliases": ["install"],
      "schema": "./src/commands/ng-add/schema.json",
      "factory": "./src/commands/ng-add/index#ngAdd",
      "description": "Add @rx-angular/state to the project."
    }
  },
  "required": [],
  "additionalProperties": false
}
