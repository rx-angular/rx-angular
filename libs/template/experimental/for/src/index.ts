export { ForModule } from './lib/for.module';
export { RxFor } from './lib/for.directive';
export { RxForViewContext } from './lib/for-view-context';
