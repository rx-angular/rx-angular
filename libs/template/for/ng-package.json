{
  "$schema": "../../../node_modules/ng-packagr/package.schema.json",
  "lib": {
    "entryFile": "src/index.ts",
    "flatModuleFile": "template-for",
    "umdModuleIds": {
      "@rx-angular/cdk/coercing": "rx-angular.cdk.coercing",
      "@rx-angular/cdk/template": "rx-angular.cdk.template",
      "@rx-angular/cdk/render-strategies": "rx-angular.cdk.render-strategies"
    }
  }
}
