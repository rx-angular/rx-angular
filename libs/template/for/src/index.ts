export { RxFor } from './lib/for.directive';
export { ForModule } from './lib/for.module';
export { RxForViewContext } from './lib/for-view-context';
