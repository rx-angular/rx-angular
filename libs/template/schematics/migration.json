{
  "$schema": "../../../node_modules/@angular-devkit/schematics/collection-schema.json",
  "schematics": {
    "update-1.0.0-beta.30": {
      "version": "1.0.0-beta.30",
      "description": "Migrate to sub-entry points",
      "factory": "./src/migrations/update-1.0.0-beta.30/index"
    },
    "introduce-rxfor-stable": {
      "version": "1.0.0-rc.0",
      "description": "Migrate rxFor imports from experimental package",
      "factory": "./src/migrations/introduce-rxfor-stable/index"
    }
  }
}
