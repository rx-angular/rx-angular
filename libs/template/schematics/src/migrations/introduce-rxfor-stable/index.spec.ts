import { Tree } from '@angular-devkit/schematics';
import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import * as path from 'path';

describe('Template Migration introduce-rxfor-stable', () => {
  let appTree: UnitTestTree;

  it('should replace module specifier', async () => {
    appTree = await setupTestFile(`
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';
      import {
        ForModule,
        RxFor
      } from '@rx-angular/template/experimental/for';
      import { RxForViewContext } from '@rx-angular/template/experimental/for';

      import { AppComponent } from './app.component';

      @NgModule({
        declarations: [
          AppComponent,
        ],
        imports: [
          BrowserModule,
          ForModule
        ],
        providers: [],
        bootstrap: [AppComponent]
      })
      export class AppModule { }
  `);

    const file = appTree.readContent('app.module.ts');

    expect(file).toMatchSnapshot();
  });

  function setupTestFile(fileInput: string, filePath = './app.module.ts') {
    const runner = new SchematicTestRunner(
      '@rx-angular/template',
      path.join(__dirname, '../../../migration.json')
    );
    const tree = new UnitTestTree(Tree.empty());

    tree.create(filePath, fileInput);

    return runner
      .runSchematicAsync(`introduce-rxfor-stable`, {}, tree)
      .toPromise();
  }
});
