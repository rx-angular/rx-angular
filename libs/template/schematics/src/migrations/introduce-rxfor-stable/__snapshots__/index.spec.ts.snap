// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template Migration introduce-rxfor-stable should replace module specifier 1`] = `
"import { ForModule, RxFor, RxForViewContext } from '@rx-angular/template/for';

import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, ForModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;
